lib/miq_automation_engine/models/miq_ae_domain.rb

```diff
@@ -40,37 +48,36 @@ def default_priority
   end
 
   def lock_contents!
-    self.system = true
+    return if source == USER_LOCKED_SOURCE # already locked
+    raise MiqAeException::CannotLock, "Cannot lock non user domains" unless source == USER_SOURCE
+    self.source = USER_LOCKED_SOURCE
     save!
   end
 
   def unlock_contents!
-    self.system = false
+    return if source == USER_SOURCE # already unlocked
+    raise MiqAeException::CannotUnlock, "Cannot unlock non user domains" unless source == USER_LOCKED_SOURCE
+    self.source = USER_SOURCE
     save!
   end
 
   def contents_locked?
-    system
+    LOCKED_SOURCES.include?(source)
   end
 
   def lockable?
-    editable_properties? && !contents_locked?
+    editable_properties? && !contents_locked? && source == USER_SOURCE
   end
 
   def unlockable?
-    editable_properties? && contents_locked?
+    editable_properties? && contents_locked? && source == USER_LOCKED_SOURCE
   end
 
   def editable_properties?
-    # TODO: In the new design will use SOURCE != SYSTEM
-    name != 'ManageIQ'
+    source != SYSTEM_SOURCE
   end
 
-  def editable_contents?(user = User.current_user)
-    # TODO: In the new design will use SOURCE != SYSTEM
-    return false if name == 'ManageIQ'
-    editable?(user)
-  end
+  alias editable_contents? editable?
```

Comment:
Any way we can avoid the alias?  I'd rather we didn't carry multiple method with the same name for no reason...If you want to deprecate one, that's fine too, but I assume most callers are internal?